kademlia/src/main/java/ingsw/group1/kademlia/NodesRoutingTable.java

```diff
@@ -0,0 +1,169 @@
+     *
+     * @param nodeOwner {@link PeerNode} node owner of routing table
+     */
+    public NodesRoutingTable(PeerNode nodeOwner) {
+        new NodesRoutingTable(nodeOwner, nodeOwner.keyLength());
+    }
+
+    /**
+     * Constructor where the routing table length is sizeTable
+     *
+     * @param nodeOwner {@link PeerNode} node owner of routing table
+     * @param sizeTable dimension of routing table
+     */
+    public NodesRoutingTable(PeerNode nodeOwner, int sizeTable) {
+        this.nodeOwner = nodeOwner;
+        this.sizeTable = sizeTable;
+        bucketsTable = new ArrayList<>(sizeTable);
+        for (int i = 0; i < sizeTable; i++)
+            bucketsTable.add(new KBucket(sizeTable));
+    }
+
+    /**
+     * @return {@link PeerNode} that is node owner of this routing table
+     */
+    public PeerNode getNodeOwner() {
+        return nodeOwner;
+    }
+
+    /**
+     * @param node {@link Node<BinarySet>} to add
+     * @return true if the node has been added, false otherwise
+     */
+    @Override
+    public boolean add(Node<BinarySet> node) {
+        if (node == null)
+            return false;
+        return bucketsTable.get(getLocation(node)).add(node);
+    }
+
+    /**
+     * @param node {@link Node<BinarySet>} to remove
+     * @return true if the node has been removed, false otherwise
+     */
+    @Override
+    public boolean remove(Node<BinarySet> node) {
+        if (node == null)
+            return false;
+        int positionNode = getLocation(node);
+        KBucket bucketFound = bucketsTable.get(positionNode);
+        if (!bucketFound.contains(node))
+            return false;
+        return bucketFound.remove(node);
+    }
+
+    /**
+     * @param node {@link Node<BinarySet>} of which check presence
+     * @return true if present, false otherwise
+     */
+    @Override
+    public boolean contains(Node<BinarySet> node) {
+        if (node == null || bucketsTable.isEmpty())
+            return false;
+        int positionNode = getLocation(node);
+        if (positionNode == -1)
+            return false;
+        return bucketsTable.get(positionNode).contains(node);
+    }
+
+    /**
+     * @param i index of the bucket in buckets container
+     * @return the bucket at index i, null otherwise
+     */
+    @Override
+    public KBucket getBucket(int i) {
+        try {
+            return bucketsTable.get(i);
+        } catch (IndexOutOfBoundsException e) {
+            return null;
+        }
+    }
+
+    /**
+     * @param node {@link Node<BinarySet>} to return its position
+     * @return the index (between 0 and N -1) of the bucket that maybe containing the given node argument, -1 otherwise
+     */
+    @Override
+    public int getLocation(Node<BinarySet> node) {
+        if (node == null)
+            return -1;
+        BinarySet distance = nodeOwner.getDistance(node);
+        return (sizeTable - 1 - distance.getFirstPositionOfOne());
+    }
+
+    /**
```

Comment:
This method is a get so I didn't write description because description is in the `@return`.
But if the description makes it more understandable I add it immediately.